src/crop-manager.sol

```diff
@@ -93,60 +93,63 @@ contract CropJoinManager {
 
 contract CropJoinManagerImp {
     bytes32 slot0;
-    mapping (address => address) public proxy;  // UrnProxy per user
+    mapping (address => address) proxy;  // UrnProxy per user
 
     address public immutable vat;
     constructor(address vat_) public {
         vat = vat_;
     }
 
-    function getProxy(address usr) internal returns (address urp) {
+    function getOrCreateProxy(address usr) public returns (address urp) {
         urp = proxy[usr];
         if (urp == address(0)) {
             urp = proxy[usr] = address(new UrnProxy(address(vat), usr));
         }
     }
 
-    function createProxy() external returns (address) {
-        return getProxy(msg.sender);
-    }
-
     function join(address crop, address urn, uint256 val) external {
         TokenLike(CropLike(crop).gem()).transferFrom(msg.sender, address(this), val);
         TokenLike(CropLike(crop).gem()).approve(crop, val);
-        CropLike(crop).join(getProxy(urn), urn, val);
+        CropLike(crop).join(getOrCreateProxy(urn), urn, val);
     }
 
     function exit(address crop, address usr, uint256 val) external {
-        CropLike(crop).exit(getProxy(msg.sender), usr, val);
+        address urp = proxy[msg.sender];
+        require(urp != address(0), "CropJoinManager/non-existing-urp");
+        CropLike(crop).exit(urp, usr, val);
     }
 
     function flee(address crop) external {
-        CropLike(crop).flee(getProxy(msg.sender));
+        address urp = proxy[msg.sender];
+        require(urp != address(0), "CropJoinManager/non-existing-urp");
+        CropLike(crop).flee(urp);
     }
 
-    function frob(address crop, address u, address v, address w, int256 dink, int256 dart) external {
-       require(u == msg.sender && v == msg.sender && w == msg.sender, "CropJoinManager/not-allowed");
+    function frob(address crop, bytes32 ilk, address u, address v, address w, int256 dink, int256 dart) external {
+        crop;
+        require(u == msg.sender && v == msg.sender && w == msg.sender, "CropJoinManager/not-allowed");
+        // require(ilk == CropLike(crop).ilk(), "CropJoinManager/non-matching-ilk"); Not needed in cur implementation
 
         // Note: This simplification only works because of the u == v == msg.sender restriction above
-        address urp = getProxy(u);
-        VatLike(vat).frob(CropLike(crop).ilk(), urp, urp, w, dink, dart);
+        address urp = getOrCreateProxy(u);
+        VatLike(vat).frob(ilk, urp, urp, w, dink, dart);
     }
 
-    function flux(address crop, address src, address dst, uint256 wad) external {
+    function flux(address crop, bytes32 ilk, address src, address dst, uint256 wad) external {
         require(src == msg.sender, "CropJoinManager/not-allowed");
+        require(ilk == CropLike(crop).ilk(), "CropJoinManager/non-matching-ilk");
```

Comment:
Since we need to perform this SLOAD regardless, does it make sense to omit the `ilk` in the arguments like we had before? My understanding was that we wanted to specify the ilk to save loading the `ilk` from the `crop`. We do get gas savings in the `frob` for now, but I think it's cleaner to just omit it.

Reply:
I thought it was more for consistency w/the Vat calls than anything. Also, it's not an SLOAD b/c `ilk` is immutable in the adapter (it is still an external call). Anyway, not something I feel strongly about either way. If Gonzalo doesn't object, I'm fine with omitting the `ilk` argument because it doesn't really seem to have any benefit.

Reply:
`ilk` removed.